packages/devtools_app/lib/src/screens/inspector_v2/inspector_controls.dart

```diff
@@ -0,0 +1,132 @@
+import '../../shared/globals.dart';
+import 'inspector_controller.dart';
+import 'inspector_screen.dart';
+
+/// Control buttons for the inspector panel.
+class InspectorControls extends StatelessWidget {
+  const InspectorControls({
+    super.key,
+    required this.controller,
+  });
+
+  final InspectorController controller;
+
+  static const serviceExtensionButtonsIncludeTextWidth = 1200.0;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        ValueListenableBuilder<bool>(
+          valueListenable: serviceConnection
+              .serviceManager.serviceExtensionManager
+              .hasServiceExtension(
+            extensions.toggleSelectWidgetMode.extension,
+          ),
+          builder: (_, selectModeSupported, __) {
+            return ServiceExtensionButtonGroup(
+              extensions: [
+                selectModeSupported
+                    ? extensions.toggleSelectWidgetMode
+                    : extensions.toggleOnDeviceWidgetInspector,
+              ],
+              minScreenWidthForTextBeforeScaling:
+                  InspectorScreenBodyState.minScreenWidthForTextBeforeScaling,
+            );
+          },
+        ),
+        const Spacer(),
+        HideImplementationWidgetsButton(controller: controller),
+        const SizedBox(width: defaultSpacing),
+        const InspectorServiceExtensionButtonGroup(),
+      ],
+    );
+  }
+}
+
+/// Group of service extension buttons for the inspector panel that control the
+/// overlays painted on the connected app.
+class InspectorServiceExtensionButtonGroup extends StatelessWidget {
+  const InspectorServiceExtensionButtonGroup({
+    super.key,
+  });
+
+  static const serviceExtensionButtonsIncludeTextWidth = 1200.0;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      children: [
+        ServiceExtensionButtonGroup(
+          minScreenWidthForTextBeforeScaling:
+              serviceExtensionButtonsIncludeTextWidth,
+          extensions: [
+            extensions.slowAnimations,
+            extensions.debugPaint,
+            extensions.debugPaintBaselines,
+            extensions.repaintRainbow,
+            extensions.invertOversizedImages,
+          ],
+        ),
+        const SizedBox(width: defaultSpacing),
+        SettingsOutlinedButton(
+          gaScreen: gac.inspector,
+          gaSelection: gac.inspectorSettings,
+          tooltip: 'Flutter Inspector Settings',
+          onPressed: () {
+            unawaited(
+              showDialog(
+                context: context,
+                builder: (context) => const FlutterInspectorSettingsDialog(),
+              ),
+            );
+          },
+        ),
+      ],
+    );
+  }
+}
+
+/// Toggle button that allows hiding the implementation widgets in the widget
+/// tree.
+class HideImplementationWidgetsButton extends StatelessWidget {
```

Comment:
I recommend flipping the logic here to phrase this in the positive. So instead of a "hide" button that defaults to false, use a "show" button that defaults to true. This is easier to reason about instead of having to compute a double negative. This follows UX guidance we've been given in the past for settings on other pages (one example is the More Debugging Options menu on the performance page).

Reply:
Sounds good, switched to "Show implementation widgets"